dex/src/main/scala/com/wavesplatform/dex/Matcher.scala

```diff
@@ -309,29 +308,44 @@ class Matcher(settings: MatcherSettings)(implicit val actorSystem: ActorSystem)
     actorSystem.actorOf(
       Props(
         new AddressDirectory(
-          wavesBlockchainAsyncClient.spendableBalanceChanges,
-          settings,
           orderDB,
-          (address, startSchedules) =>
-            Props(
-              new AddressActor(
-                address,
-                asset => wavesBlockchainAsyncClient.spendableBalance(address, asset),
-                time,
-                orderDB,
-                wavesBlockchainAsyncClient.forgedOrder,
-                matcherQueue.storeEvent,
-                orderBookCache.getOrDefault(_, OrderBook.AggregatedSnapshot()),
-                startSchedules,
-                settings.actorResponseTimeout - settings.actorResponseTimeout / 10 // Should be enough
-              )
-          ),
+          createAddressActor,
           historyRouter
         )
       ),
       "addresses"
     )
 
+  private val spendableBalancesActor = {
+    actorSystem.actorOf(
+      Props(
+        new SpendableBalancesActor(
+          wavesBlockchainAsyncClient.spendableBalances,
+          wavesBlockchainAsyncClient.allAssetsSpendableBalance,
+          addressActors
+        )
+      )
+    ) unsafeTap { sba =>
+      wavesBlockchainAsyncClient.spendableBalanceChanges.foreach(sba ! SpendableBalancesActor.Command.UpdateStates(_))(monixScheduler)
+    }
+  }
```

Comment:
`val` creates an instance eagerly. Why not just write:
```scala
wavesBlockchainAsyncClient.spendableBalanceChanges.foreach(spendableBalancesActor ! SpendableBalancesActor.Command.UpdateStates(_))(monixScheduler)
```
right after `spendableBalancesActor`? Just interesting :)

Reply:
Don't like just side effect snippets